validator/src/commands/exit/mod.rs

```diff
@@ -101,17 +125,89 @@ pub fn execute(matches: &ArgMatches, ledger_path: &Path) -> Result<()> {
         )?;
     }
 
-    let admin_client = admin_rpc_service::connect(ledger_path);
-    admin_rpc_service::runtime().block_on(async move { admin_client.await?.exit().await })?;
+    // Grab the pid from the process before initiating exit as the running
+    // validator will be unable to respond after exit has returned.
+    //
+    // Additionally, delay checking the result until it will actually be used.
+    // In an upgrade scenario, it is possible that a binary that calls pid()
+    // will be iniating exit against a process that doesn't support pid().
+    // Since PostExitAction::Wait case is opt-in (via --wait-for-exit), the
+    // result is checked ONLY in that case to provide a friendlier upgrade
+    // path for user who are NOT using --wait-for-exit
+    let validator_pid_result = admin_rpc_service::runtime().block_on(async move {
+        let admin_client = admin_rpc_service::connect(ledger_path).await?;
+        let validator_pid_result = admin_client.pid().await;
```

Comment:
can we match here and give better guidance to the caller? something like,
```
"error: remote process exit cannot be waited. `--wait-for-exit` is not supported."
```

Reply:
ea1c9d0cdfe5db69bd60eb2aff8c5afd45110f84

The final error output now looks like this:
```
Validator command failed: remote process exit cannot be waited on. `--wait-for-exit` is not supported by the remote process
```